apps/userdashboard/api.py

```diff
@@ -84,6 +84,17 @@ def update(self, request, *args, **kwargs):
 
         return super().update(request, *args, **kwargs)
 
+    @action(detail=True)
+    def toggle_show_ai_report(self, request, **kwargs):
+        comment = self.get_object()
+        if hasattr(comment, "ai_report") and comment.ai_report:
+            comment.ai_report.show_in_discussion = (
```

Comment:
here you need ome these two checks, not both, as they would both return true if a comment has an ai_report attached to it.

Reply:
I agree only the first check is needed (unless we use select_prefetch) somewhere, then the ai_report attribute always exists but is None (hence the second check).